sublime-package.json

```diff
@@ -394,6 +394,32 @@
             }
           }
         }
+      },
+      {
+        "file_patterns": [
+          "/*.sublime-project"
+        ],
+        "schema": {
+          "properties": {
+            "settings": {
+              "properties": {
+                "LSP": {
+                  "type": "object",
+                  "markdownDescription": "The dictionary of your configured language servers or overrides for existing configurations. The keys of this dictionary are free-form. They give a humany-friendly name to the server configuration. They are shown in the bottom-left corner in the status bar once attached to a view (unless you have `\"show_view_status\"` set to `false`).",
+                  "additionalProperties": {
+                    "$ref": "sublime://settings/LSP#/definitions/ClientConfig"
+                  },
+                },
+                "lsp_format_on_save": {
+                  "$ref": "sublime://settings/LSP#/definitions/lsp_format_on_save",
+                },
+                "lsp_code_actions_on_save": {
+                  "$ref": "sublime://settings/LSP#/definitions/lsp_code_actions_on_save"
```

Comment:
Does this mean everyone has to write the same pieces of json for both the `/Preferences.sublime-settings` pattern as well as the `*.sublime-settings` pattern? I would prefer if this was automated in LSP-json.

Reply:
Do you mean `Preferences.sublime-settings` and `*.sublime-project`?

Note that in this case those are not the same. The LSP object is unique.

But yeah, ideally the global settings should be handled automatically somehow but I'm not sure how. Maybe we could automatically attach schemas to file patterns based on `$id` matching some pattern.

Reply:
> Note that in this case those are not the same. The LSP object is unique.

True. So there would be three file patterns: `/LSP.sublime-settings`, `/Preferences.sublime-settings` and `*.sublime-project`. The schema from `/Preferences.sublime-settings` should be merged into `*.sublime-project`, together with anything from `*.sublime-project` itself (in this case, only this `LSP` object).

Reply:
Maybe you can let the schema author define what should be merged into what? For example, an array of `"mergeWith": ["/Preferences.sublime-settings"]` for `*.sublime-project`. I'm not sure if it's technically possible to do such things because I'm not familiar with the code from LSP-json.

Reply:
Are you thinking to have that `mergeWith` at the root level of the schema or possible within any object type?
If the former then there would still need to be some custom logic for specific files to know that properties from `Preferences.json` schema go into the `settings` object. And in both cases, the schemas would need to be manually altered in-memory to add `allOf` or such. It sounds like a bit of a pain.

Reply:
I have some code to merge those things so it should be doable.

Reply:
Yeah, LSP-json has to do this automatically because otherwise this will be a major roadblock for other plugin devs to adopt sublime-package.json. I also predict not many plugin devs actually fully understand the override behavior of ST config files. So if LSP-json does this automatically, devs learn something by accident (the best kind of learning).

Reply:
This should address the issue: https://github.com/sublimelsp/LSP-json/pull/46